stl/inc/ranges

```diff
@@ -1627,6 +1627,124 @@ namespace ranges {
         using all_t = decltype(all(_STD declval<_Rng>()));
     } // namespace views
 
+#if _HAS_CXX23
+    template <input_range _Vw>
+        requires view<_Vw>
+    class as_rvalue_view : public view_interface<as_rvalue_view<_Vw>> {
+    private:
+        /* [[no_unique_address]] */ _Vw _Range{};
+
+        template <range _Rng>
+        static constexpr bool _Is_end_nothrow_v = noexcept(move_sentinel{_RANGES end(_STD declval<_Rng&>())});
+
+        template <common_range _Rng>
+        static constexpr bool _Is_end_nothrow_v<_Rng> = noexcept(move_iterator{_RANGES end(_STD declval<_Rng&>())});
+
+    public:
+        // clang-format off
+        as_rvalue_view() requires default_initializable<_Vw> = default;
+        // clang-format on
+
+        constexpr explicit as_rvalue_view(_Vw _Range_) noexcept(is_nothrow_move_constructible_v<_Vw>) // strengthened
+            : _Range(_STD move(_Range_)) {}
+
+        _NODISCARD constexpr _Vw base() const& noexcept(is_nothrow_copy_constructible_v<_Vw>) // strengthened
+            requires copy_constructible<_Vw> {
+            return _Range;
+        }
+
+        _NODISCARD constexpr _Vw base() && noexcept(is_nothrow_move_constructible_v<_Vw>) /* strengthened */ {
+            return _STD move(_Range);
+        }
+
+        _NODISCARD constexpr auto begin() noexcept(noexcept(move_iterator{_RANGES begin(_Range)})) // strengthened
+            requires(!_Simple_view<_Vw>) {
+            return move_iterator{_RANGES begin(_Range)};
+        }
+
+        _NODISCARD constexpr auto begin() const noexcept(noexcept(move_iterator{_RANGES begin(_Range)})) // strengthened
+            requires range<const _Vw> {
+            return move_iterator{_RANGES begin(_Range)};
+        }
+
+        _NODISCARD constexpr auto end() noexcept(_Is_end_nothrow_v<_Vw>) // strengthened
+            requires(!_Simple_view<_Vw>) {
+            if constexpr (common_range<_Vw>) {
+                return move_iterator{_RANGES end(_Range)};
+            } else {
+                return move_sentinel{_RANGES end(_Range)};
+            }
+        }
+
+        _NODISCARD constexpr auto end() const noexcept(_Is_end_nothrow_v<const _Vw>) // strengthened
+            requires range<const _Vw> {
+            if constexpr (common_range<const _Vw>) {
+                return move_iterator{_RANGES end(_Range)};
+            } else {
+                return move_sentinel{_RANGES end(_Range)};
+            }
+        }
+
+        _NODISCARD constexpr auto size() noexcept(noexcept(_RANGES size(_Range))) // strengthened

```

Comment:
```suggestion
        _NODISCARD constexpr auto size() noexcept(noexcept(_Fake_decay_copy(_RANGES size(_Range)))) // strengthened
```

we decay-copy the return value of `_RANGES size()`. Same below.

Reply:
@CaseyCarter please confirm? We don't do this elsewhere and I think it's a bug.

Reply:
`ranges::size` always returns a prvalue.

Reply:
And (as we discussed at the maintainer meeting) integer-like types don't throw exceptions - so only the actual call to `ranges::size` can throw.